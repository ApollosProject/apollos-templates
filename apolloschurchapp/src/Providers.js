import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers, NavigationService } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import {
  LiveProvider,
  ACCEPT_FOLLOW_REQUEST,
} from '@apollosproject/ui-connected';
import { checkOnboardingStatusAndNavigate } from '@apollosproject/ui-onboarding';
import { ONBOARDING_VERSION } from './ui/Onboarding';

import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      // TODO deprecated prop
      navigate={NavigationService.navigate}
      handleExternalLink={(url) => {
        const path = url.split('app-link/')[1];
        const [route, location] = path.split('/');
        if (route === 'content')
          NavigationService.navigate('ContentSingle', { itemId: location });
        if (route === 'nav')
          NavigationService.navigate(
            // turns "home" into "Home"
            location[0].toUpperCase() + location.substring(1)
          );
      }}
      actionMap={{
        // accept a follow request when someone taps "accept" in a follow request push notification
        acceptFollowRequest: ({ requestPersonId }) =>
          client.mutate({
            mutation: ACCEPT_FOLLOW_REQUEST,
            variables: { personId: requestPersonId },
          }),
      }}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() =>
          checkOnboardingStatusAndNavigate({
            client,
            navigation: NavigationService,
            latestOnboardingVersion: ONBOARDING_VERSION,
          })
        }
      >
        <AnalyticsProvider>
          <LiveProvider>
            <Providers
              themeInput={customTheme}
              iconInput={customIcons}
              {...props}
            />
          </LiveProvider>
        </AnalyticsProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
