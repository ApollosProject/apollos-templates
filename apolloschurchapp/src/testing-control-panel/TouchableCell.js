import React from 'react';
import PropType from 'prop-types';
import { Touchable, Cell, CellIcon, CellText } from '@apollosproject/ui-kit';

const TouchableCell = ({ iconName, cellText, handlePress }) => (
  <Touchable onPress={handlePress}>
    <Cell>
      <CellIcon name={iconName} />
      <CellText>{cellText}</CellText>
    </Cell>
  </Touchable>
);

TouchableCell.propTypes = {
  iconName: PropType.string.isRequired,
  cellText: PropType.string.isRequired,
  handlePress: PropType.func.isRequired,
};

export default TouchableCell;
