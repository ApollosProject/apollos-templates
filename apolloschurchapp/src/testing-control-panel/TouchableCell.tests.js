import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../Providers';

import TouchableCell from './TouchableCell';

describe('touchable cell', () => {
  it('renders the cell, icon, and text', () => {
    const tree = renderer.create(
      <Providers>
        <TouchableCell
          cellText="This is a cell"
          iconName="play"
          handlePress={() => ({})}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
