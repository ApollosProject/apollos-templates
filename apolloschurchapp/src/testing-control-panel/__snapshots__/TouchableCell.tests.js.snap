// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`touchable cell renders the cell, icon, and text 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "minHeight": 40,
          "padding": 8,
          "paddingLeft": 16,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "paddingLeft": 0,
            },
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={20}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M19.7793 10.4688L6.28223 2.22027C6.05488 2.08125 5.79461 2.00531 5.52817 2.00027C5.26174 1.99523 4.99878 2.06126 4.76634 2.19158C4.5339 2.3219 4.34038 2.51179 4.20568 2.74173C4.07099 2.97166 4 3.23333 4 3.49981V19.9983C4.00022 20.2647 4.07135 20.5263 4.20609 20.7561C4.34083 20.9859 4.53433 21.1757 4.76669 21.3059C4.99906 21.4362 5.26192 21.5023 5.52826 21.4974C5.79461 21.4924 6.05484 21.4167 6.28223 21.2779L19.7793 13.0294C19.9986 12.8953 20.1797 12.7071 20.3054 12.483C20.4311 12.2588 20.4971 12.0061 20.4971 11.7491C20.4971 11.4921 20.4311 11.2394 20.3054 11.0152C20.1797 10.791 19.9986 10.6029 19.7793 10.4688V10.4688Z"
            fill={
              Array [
                0,
                4281348144,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#303030",
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": null,
            "fontWeight": null,
            "lineHeight": 21,
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        This is a cell
      </Text>
    </View>
  </View>,
]
`;
